tests/kernel/wave/wave_e2e_test.py

```diff
@@ -4,27 +4,34 @@
 import torch
 from numpy.testing import assert_allclose
 import pytest
+import sympy
 import os
 
 _run_e2e = int(os.environ.get("WAVE_RUN_E2E_TESTS", 0))
 
 require_e2e = pytest.mark.skipif(not _run_e2e, reason="e2e tests are disabled")
 
 
+_test_shapes = [(1, 128), (256, 64), (256, 128), (256, 256), (256, 1024)]
+
+
 @require_e2e
-def test_copy():
+@pytest.mark.parametrize("shape", _test_shapes)
+def test_copy(shape):
     M = tkl.sym.M
     N = tkl.sym.N
-    BLOCK_M = tkl.sym.BLOCK_M
-    BLOCK_N = tkl.sym.BLOCK_N
-    ELEMS_PER_THREAD = tkl.sym.ELEMS_PER_THREAD
     ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
 
+    wave_size = 64
+    BLOCK_M = 1
+    BLOCK_N = sympy.Min(N, 256)
+    ELEMS_PER_THREAD = BLOCK_N / wave_size

```

Comment:
Since wave_e2e_test.py may be a collection of reference kernels for people to use, would be good to set some comments on why we select some of these.

For example, we can comment that `ELEMS_PER_THREAD = BLOCK_N / wave_size` is selected because we are distributing BLOCK_N to a single subgroup.

Reply:
Added comment